dhis-2/dhis-services/dhis-service-administration/src/test/java/org/hisp/dhis/maintenance/MaintenanceServiceTest.java

```diff
@@ -210,6 +210,40 @@ public void setUpTest()
         programStageInstanceService.addProgramStageInstance( programStageInstanceWithTeiAssociation );
     }
 
+    @Test
+    void testDeleteSoftDeletedTrackedEntityInstanceLinkedToARelationshipItem()
+    {
+        RelationshipType rType = createRelationshipType( 'A' );
+        rType.getFromConstraint().setRelationshipEntity( RelationshipEntity.PROGRAM_INSTANCE );
+        rType.getFromConstraint().setProgram( program );
+        rType.getToConstraint().setRelationshipEntity( RelationshipEntity.TRACKED_ENTITY_INSTANCE );
+        rType.getFromConstraint().setTrackedEntityType( entityInstance.getTrackedEntityType() );
+        relationshipTypeService.addRelationshipType( rType );
+        Relationship r = new Relationship();
+        RelationshipItem rItem1 = new RelationshipItem();
+        rItem1.setProgramInstance( programInstance );
+        RelationshipItem rItem2 = new RelationshipItem();
+        rItem2.setTrackedEntityInstance( entityInstance );
+        r.setFrom( rItem1 );
+        r.setTo( rItem2 );
+        r.setRelationshipType( rType );
+        r.setKey( RelationshipUtils.generateRelationshipKey( r ) );
+        r.setInvertedKey( RelationshipUtils.generateRelationshipInvertedKey( r ) );
```

Comment:
Could you please group this and the other tests into 3 blocks (arrange, act, assert). Its not obvious what method is being tested.